graylog2-web-interface/src/pages/ClusterConfigurationPage.tsx

```diff
@@ -0,0 +1,97 @@
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import { DocumentTitle, Icon, PageHeader } from 'components/common';
+import { Col, Row, SegmentedControl } from 'components/bootstrap';
+import useClusterNodes from 'components/cluster-configuration/useClusterNodes';
+import ClusterConfigurationGraphView from 'components/cluster-configuration/ClusterConfigurationGraphView';
+import ClusterConfigurationListView from 'components/cluster-configuration/ClusterConfigurationListView';
+import TableFetchContextProvider from 'components/common/PaginatedEntityTable/TableFetchContextProvider';
+import type { SearchParams } from 'stores/PaginationTypes';
+import ClusterConfigurationPageNavigation from 'components/cluster-configuration/ClusterConfigurationPageNavigation';
+import HideOnCloud from 'util/conditional/HideOnCloud';
+import IndexerClusterHealth from 'components/indexers/IndexerClusterHealth';
+
+const ViewTypeSwitchContainer = styled(Col)`
+  display: flex;
+  justify-content: right;
+`;
+
+const VIEW_TYPES_SEGMENTS = [
+  {
+    value: 'list' as const,
+    label: (<Icon name="list" />),
+  },
+  {
+    value: 'cards' as const,
+    label: (<Icon name="account_tree" type='regular' />),
+  },
+];
+
+type ViewTypesSegments = 'list' | 'cards';
+
+const ClusterConfigurationPage = () => {
+  const [viewType, setViewType] = useState<ViewTypesSegments>('list');
+  const clusterNodes = useClusterNodes();
+  const searchParams: SearchParams = { query: '', page: 1, pageSize: 0, sort: { attributeId: 'hostname', direction: 'asc' } };
+
+  return (
+    <DocumentTitle title="Cluster Configuration">
+      <ClusterConfigurationPageNavigation />
+      <div>
+        <PageHeader title="Cluster Configuration">
+          <span>
+            This page provides a real-time overview of the nodes in your Graylog cluster.
+            You can pause message processing at any time. The process buffers will not accept any new messages until
+            you resume it. If the message journal is enabled for a node, which it is by default, incoming messages
+            will be persisted to disk, even when processing is disabled.
+          </span>
+        </PageHeader>
+        <HideOnCloud>
+          <IndexerClusterHealth minimal />
+        </HideOnCloud>
+        <Row className="content">
+          <Col xs={6}>
+            <h2>Nodes</h2>
+          </Col>
+          {false && (
+            <ViewTypeSwitchContainer xs={6}>
+              <SegmentedControl data={VIEW_TYPES_SEGMENTS}
+                                radius="sm"
+                                value={viewType}
+                                onChange={(newViewType) => setViewType(newViewType)} />
+            </ViewTypeSwitchContainer>
+          )}
+          <Col md={12}>
+            <TableFetchContextProvider refetch={clusterNodes.refetchDatanodes} searchParams={searchParams} attributes={[]}>
```

Comment:
Since we want to show the Graylog and data-node nodes on a single page, we should create a proper endpoint that handles the entities. That way, we can use the EntityDataTable and have pagination and other cool features. Is that something you’re planning?

Reply:
Yes that's the end goal: to have a proper endpoint for all the cluster's nodes that includes also MongoDB instances, forwarders and maybe also sidecars, but for this cycle the goal was the visual (frontend part) of a centralized Cluster Configuration page only. We will likely add that endpoint next cycle.